module/proto/peggy/v1/batch.proto

```diff
@@ -23,3 +23,14 @@ message OutgoingTransferTx {
   ERC20Token erc20_token  = 4;
   ERC20Token erc20_fee    = 5;
 }
+
+// OutgoingLogicCall represents an individual logic call from Peggy to ETH
+message OutgoingLogicCall {
```

Comment:
Maybe I'm missing something here, but this doesn't seem to conform to the actual arguments for a logic call. Here they are for reference:

```
	struct LogicCallArgs {
		// Transfers out to the logic contract
		uint256[] transferAmounts;
		address[] transferTokenContracts;
		// The fees (transferred to msg.sender)
		uint256[] feeAmounts;
		address[] feeTokenContracts;
		// The arbitrary logic call
		address logicContractAddress;
		bytes payload;
		// Invalidation metadata
		uint256 timeOut;
		bytes32 invalidationId;
		uint256 invalidationNonce;
	}
```

First of all, a lot of them are named something different. Why?

I'm guessing "logic_nonce" is supposed to be the invalidation_nonce, and "call" is supposed to be the payload? I'm guessing "id" is supposed to be the invalidation_id? If so, it is the wrong type. It should be []bytes.

And we're totally missing "transfer_amounts". How can the e.g., the Uniswap liquidity redeeming logic redeem LP tokens if Gravity doesn't give it any LP tokens?

Reply:
I'll just write out what I think this should probably be.

```
message OutgoingLogicCall {
  repeated ERC20Token transfers     = 1;
  repeated ERC20Token fees     = 2;
  string logic_contract_address = 3;
  bytes payload = 4;
  uint64 timeout = 5;
  bytes invalidation_id = 6;
  unit64 invalidation_nonce = 7;
}
```

Reply:
I will say that we might be able to name some of the arguments for the contract method something else. For example "invalidation_id" could definitely be changed to "invalidation_scope". If we decide to do that, I'll change it in the solidity as well.

Reply:
I'm fine with names as is. Will refactor to match here